src/read_text_file.cpp

```diff
@@ -0,0 +1,25 @@
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <R.h>
+#include <Rinternals.h>
+
+extern "C" SEXP read_text_file(SEXP path) {
+  std::ifstream t(CHAR(asChar(path)));
+  std::string s;
+  std::getline(t, s);
+  SEXP out = PROTECT(allocVector(STRSXP, 1));
+  SET_STRING_ELT(out, 0, mkChar(s.c_str()));
+  UNPROTECT(1);
+  return out;
+}
+
+static const R_CallMethodDef callMethods[] = {
+  {"read_text_file", (DL_FUNC) &read_text_file, 2},
+  {NULL, NULL, 0}
+};
+
+void R_init_storr(DllInfo *info) {
```

Comment:
See https://github.com/richfitz/ring/blob/master/src/registration.c#L58-L65 for how I deal with this usually - do `#include <Rversion.h>` earlier up the file and then make `R_useDynamicSymbols` and `R_forceSymbols` conditional on version.  Please use `call_methods` rather than `callMethods`.

Reply:
Glad I know this now. Maybe we can shoot for compatibility with R-3.3.x after all.

Reply:
See 4f06d03782e2d7bbe2ee7f35b97e0fcb893dac86.